src/project/project.service.ts

```diff
@@ -918,72 +934,84 @@ export class ProjectService {
       );
     }
 
-    const enrollment = await this.enrollmentRepository.findOne({
-      where: {
-        project: {
-          id: project.id,
-        },
-        user: {
-          id: userId,
-        },
-      },
-      select: ['id', 'requestState'],
-    });
-    if (!enrollment)
-      throw new BadRequest('Este usuario no tiene una solicitud pendiente');
+    // Update the request state
+    await this.updateRequestState(
+      action === 'approve' ? RequestState.Accepted : RequestState.Rejected,
+      project.id,
+      userRequested.id,
+      enrollment.id,
+      enrollRequestAdminDto.message,
+      action === 'approve' ? 'approved' : 'rejected',
+    );
```

Comment:
We are asking twice for action to equal 'approve', can this be avoided? 
Additionally, do we really need last parameter passed?  It is only being used by debugger.

In case we really want it to be passed, can't we make a const action only one time and then pass it wherever needed -  We would be passing a RequestState.Accepted or RequestState.Rejected instead of hardcoded string

Reply:
If I remove this parameter and directly use RequestState for the debugger message (inside the private function), this would go against the suggestion we took based on your feedback at the last meet, which was that there were unnecessarily nested conditions. Passing this parameter solved that problem since it eliminates the need to make nested ternaries (in this case, for the debug message).